Build error on Summit: undefined reference to `void adios2::helper::CUDAMinMax`

I am getting an error on Summit during compilation of v2.8.1
```[ 31%] Linking CXX executable ../../../../../bin/sst_conn_tool
../../../../../lib64/libadios2_core.so.2.8.1: undefined reference to `void adios2::helper::CUDAMinMax<long double>(long double const*, unsigned long, long double&, long double&)'
```


Compiler used: PGI 20.4
CUDA version: 11.0.3

Modules loaded:

```
  1) hsi/5.0.2.p5
  2) xalt/1.2.1
  3) DefApps
  4) cmake/3.23.1
  5) python/3.7-anaconda3
  6) git-lfs/2.11.0
  7) pgi/20.4
  8) spectrum-mpi/10.4.0.3-20210112
  9) nsight-compute/2021.2.1
 10) nsight-systems/2021.3.1.54
 11) cuda/11.0.3
 12) hypre/2.22.0-cpu
 13) lsf-tools/2.0
 14) amgx/2.1.0-1
 15) netcdf-fortran/4.4.5
 16) git/2.35.1
```

The configure log and build log are attached.
[build.txt](https://github.com/ornladios/ADIOS2/files/8960768/build.txt)
[config.txt](https://github.com/ornladios/ADIOS2/files/8960769/config.txt)

--- a/source/adios2/toolkit/format/bp/bp4/BP4Serializer.tcc
+++ b/source/adios2/toolkit/format/bp/bp4/BP4Serializer.tcc
@@ -339,7 +339,8 @@ BP4Serializer::GetBPStats(const bool singleValue,
     if (blockInfo.MemSpace == MemorySpace::CUDA)
     {
         const size_t size = helper::GetTotalSize(blockInfo.Count);
-        helper::CUDAMinMax(blockInfo.Data, size, stats.Min, stats.Max);
+        if (!std::is_same<T, long double>::value)
+            helper::CUDAMinMax(blockInfo.Data, size, stats.Min, stats.Max);
         return stats;
     }
 #endif

--- a/source/adios2/toolkit/format/bp5/BP5Serializer.cpp
+++ b/source/adios2/toolkit/format/bp5/BP5Serializer.cpp
@@ -606,8 +606,9 @@ static void GetMinMax(const void *Data, size_t ElemCount, const DataType Type,
              Type == helper::GetDataType<T>())                                 \
     {                                                                          \
         const T *values = (const T *)Data;                                     \
-        helper::CUDAMinMax(values, ElemCount, MinMax.MinUnion.field_##N,       \
-                           MinMax.MaxUnion.field_##N);                         \
+        if (!std::is_same<T, long double>::value)                              \
+            helper::CUDAMinMax(values, ElemCount, MinMax.MinUnion.field_##N,   \
+                               MinMax.MaxUnion.field_##N);                     \
     }
     ADIOS2_FOREACH_MINMAX_STDTYPE_2ARGS(pertype)
 #undef pertype

